extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js

```diff
@@ -358,3 +367,64 @@ ve.init.wikia.ViewPageTarget.prototype.maybeShowDialogs = function () {
 	}
 	//this.surface.getDialogs().getWindow( 'wikiaSingleMedia' ).open( null, null, this.surface );
 };
+
+/**
+ * @inheritdoc
+ */
+ve.init.wikia.ViewPageTarget.prototype.onSaveErrorCaptcha = function ( editApi ) {
+	var $captchaDiv = $( '<div>' ).addClass( 've-ui-mwSaveDialog-captcha' );
+
+	if ( grecaptcha ) {
+		this.captchaResponse = null;
+		this.renderReCaptcha( $captchaDiv );
+	} else {
+		this.loadAndRenderFancyCaptcha( $captchaDiv );
+	}
+
+	this.saveDialog.clearMessage( 'api-save-error' );
+	this.saveDialog.showMessage( 'api-save-error', $captchaDiv, { wrap: false } );
+	this.saveDialog.popPending();
+	this.events.trackSaveError( 'captcha' );
+};
+
+/**
+ * Render reCaptcha
+ *
+ * @method
+ * @param {jQuery} $container
+ */
+ve.init.mw.ViewPageTarget.prototype.renderReCaptcha = function ( $container ) {
+	grecaptcha.render( $container[0], {
+		sitekey: mw.config.get( 'reCaptchaPublicKey' ),
+		theme: 'light',
+		callback: function ( response ) {
+			this.captchaResponse = response;
+		}.bind( this )
+	} );
+};
+
+/**
+ * Render Fancy Captcha
+ *
+ * @method
+ * @param {jQuery} $container
+ */
+ve.init.mw.ViewPageTarget.prototype.loadAndRenderFancyCaptcha = function ( $container ) {
+	var getFancyCaptchaDeferred = $.Deferred();
+
+	$.nirvana.sendRequest( {
+		controller: 'CaptchaController',
+		method: 'getFancyCaptcha',
+		type: 'GET',
+		callback: function ( data ) {
+			getFancyCaptchaDeferred.resolve( data );
+		}
+	} );
+
+	$.when(
```

Comment:
Ah, thanks @inez! Yea, I had been looking through the code base to figure out the best way to load that scss asynchronously, but this makes a lot more sense than the solution I came up with. I'm book marking it to use as a reference. Thanks for pinging me on it!